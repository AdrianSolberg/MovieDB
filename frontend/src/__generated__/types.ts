export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
  Date: { input: any; output: any; }
};

export type Movie = {
  _id: Scalars['Int']['output'];
  title: Scalars['String']['output'];
  vote_average: Scalars['Float']['output'];
  vote_count: Scalars['Int']['output'];
  status: Scalars['String']['output'];
  release_date: Scalars['Date']['output'];
  revenue: Scalars['Float']['output'];
  runtime: Scalars['Int']['output'];
  backdrop_path?: Maybe<Scalars['String']['output']>;
  budget: Scalars['Int']['output'];
  homepage?: Maybe<Scalars['String']['output']>;
  imdb_id?: Maybe<Scalars['String']['output']>;
  original_language: Scalars['String']['output'];
  original_title: Scalars['String']['output'];
  overview: Scalars['String']['output'];
  popularity: Scalars['Float']['output'];
  poster_path?: Maybe<Scalars['String']['output']>;
  tagline?: Maybe<Scalars['String']['output']>;
  genres: Array<Scalars['String']['output']>;
  production_companies: Array<Scalars['String']['output']>;
  production_countries: Array<Scalars['String']['output']>;
  spoken_languages: Array<Scalars['String']['output']>;
  keywords: Array<Scalars['String']['output']>;
};

export type Filters = {
  Genre: Array<Scalars['String']['output']>;
  Rating: Array<Scalars['String']['output']>;
  Decade: Array<Scalars['String']['output']>;
  Status: Array<Scalars['String']['output']>;
  Runtime: Array<Scalars['String']['output']>;
};

export type FiltersInput = {
  Genre: Array<Scalars['String']['input']>;
  Rating: Array<Scalars['String']['input']>;
  Decade: Array<Scalars['String']['input']>;
  Status: Array<Scalars['String']['input']>;
  Runtime: Array<Scalars['String']['input']>;
};

export enum SortingType {
  NewestFirst = 'NEWEST_FIRST',
  OldestFirst = 'OLDEST_FIRST',
  BestRated = 'BEST_RATED',
  WorstRated = 'WORST_RATED',
  LongestRuntime = 'LONGEST_RUNTIME',
  ShortestRuntime = 'SHORTEST_RUNTIME'
}

export type Query = {
  movie?: Maybe<Movie>;
  movies: Array<Movie>;
  filters?: Maybe<Filters>;
};


export type QueryMovieArgs = {
  id: Scalars['Int']['input'];
};


export type QueryMoviesArgs = {
  skip?: InputMaybe<Scalars['Int']['input']>;
  limit?: InputMaybe<Scalars['Int']['input']>;
  filters?: InputMaybe<FiltersInput>;
  sortOption?: InputMaybe<SortingType>;
  search?: InputMaybe<Scalars['String']['input']>;
};

export type GetMovieQueryVariables = Exact<{
  movieId: Scalars['Int']['input'];
}>;


export type GetMovieQuery = { __typename?: 'Query', movie?: { __typename?: 'Movie', _id: number, title: string, vote_average: number, vote_count: number, status: string, release_date: any, revenue: number, runtime: number, backdrop_path?: string | null, budget: number, homepage?: string | null, imdb_id?: string | null, original_language: string, original_title: string, overview: string, popularity: number, poster_path?: string | null, tagline?: string | null, genres: Array<string>, production_companies: Array<string>, production_countries: Array<string>, spoken_languages: Array<string>, keywords: Array<string> } | null };

export type GetMoviesQueryVariables = Exact<{
  skip?: InputMaybe<Scalars['Int']['input']>;
  limit?: InputMaybe<Scalars['Int']['input']>;
  filters?: InputMaybe<FiltersInput>;
  sortOption?: InputMaybe<SortingType>;
  search?: InputMaybe<Scalars['String']['input']>;
}>;


export type GetMoviesQuery = { __typename?: 'Query', movies: Array<{ __typename?: 'Movie', _id: number, title: string, vote_average: number, release_date: any, runtime: number, poster_path?: string | null }> };

export type GetFiltersQueryVariables = Exact<{ [key: string]: never; }>;


export type GetFiltersQuery = { __typename?: 'Query', filters?: { __typename?: 'Filters', Genre: Array<string>, Rating: Array<string>, Decade: Array<string>, Status: Array<string>, Runtime: Array<string> } | null };
