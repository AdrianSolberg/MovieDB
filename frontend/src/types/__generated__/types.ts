export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = {
    [K in keyof T]: T[K];
};
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
    [SubKey in K]?: Maybe<T[SubKey]>;
};
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
    [SubKey in K]: Maybe<T[SubKey]>;
};
export type MakeEmpty<
    T extends { [key: string]: unknown },
    K extends keyof T,
> = { [_ in K]?: never };
export type Incremental<T> =
    | T
    | {
          [P in keyof T]?: P extends " $fragmentName" | "__typename"
              ? T[P]
              : never;
      };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
    ID: { input: string; output: string };
    String: { input: string; output: string };
    Boolean: { input: boolean; output: boolean };
    Int: { input: number; output: number };
    Float: { input: number; output: number };
    DateTime: { input: Date; output: Date };
};

export type Movie = {
    __typename?: "Movie";
    _id: Scalars["Int"]["output"];
    title: Scalars["String"]["output"];
    vote_average: Scalars["Float"]["output"];
    vote_count: Scalars["Int"]["output"];
    status: Scalars["String"]["output"];
    release_date: Scalars["DateTime"]["output"];
    revenue: Scalars["Float"]["output"];
    runtime: Scalars["Int"]["output"];
    backdrop_path?: Maybe<Scalars["String"]["output"]>;
    budget: Scalars["Int"]["output"];
    homepage?: Maybe<Scalars["String"]["output"]>;
    imdb_id?: Maybe<Scalars["String"]["output"]>;
    original_language: Scalars["String"]["output"];
    original_title: Scalars["String"]["output"];
    overview: Scalars["String"]["output"];
    popularity: Scalars["Float"]["output"];
    poster_path?: Maybe<Scalars["String"]["output"]>;
    tagline?: Maybe<Scalars["String"]["output"]>;
    genres: Array<Scalars["String"]["output"]>;
    production_companies: Array<Scalars["String"]["output"]>;
    production_countries: Array<Scalars["String"]["output"]>;
    spoken_languages: Array<Scalars["String"]["output"]>;
    keywords: Array<Scalars["String"]["output"]>;
    reviews: Array<Review>;
};

export type Review = {
    __typename?: "Review";
    _id: Scalars["ID"]["output"];
    movie: Movie;
    username: Scalars["String"]["output"];
    rating: Scalars["Int"]["output"];
    comment: Scalars["String"]["output"];
    date: Scalars["DateTime"]["output"];
};

export type Filters = {
    __typename?: "Filters";
    Genre: Array<Scalars["String"]["output"]>;
    Rating: Array<Scalars["String"]["output"]>;
    Decade: Array<Scalars["String"]["output"]>;
    Status: Array<Scalars["String"]["output"]>;
    Runtime: Array<Scalars["String"]["output"]>;
};

export type FiltersInput = {
    Genre: Array<Scalars["String"]["input"]>;
    Rating: Array<Scalars["String"]["input"]>;
    Decade: Array<Scalars["String"]["input"]>;
    Status: Array<Scalars["String"]["input"]>;
    Runtime: Array<Scalars["String"]["input"]>;
};

export enum SortingType {
    MOST_POPULAR = "MOST_POPULAR",
    NEWEST_FIRST = "NEWEST_FIRST",
    OLDEST_FIRST = "OLDEST_FIRST",
    BEST_RATED = "BEST_RATED",
    WORST_RATED = "WORST_RATED",
    LONGEST_RUNTIME = "LONGEST_RUNTIME",
    SHORTEST_RUNTIME = "SHORTEST_RUNTIME",
}

export type Query = {
    __typename?: "Query";
    movie?: Maybe<Movie>;
    movies: Array<Movie>;
    filters: Filters;
    latestReviews: Array<Review>;
    userReviews: Array<Review>;
};

export type QuerymovieArgs = {
    id: Scalars["Int"]["input"];
};

export type QuerymoviesArgs = {
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
    filters?: InputMaybe<FiltersInput>;
    sortOption?: InputMaybe<SortingType>;
    search?: InputMaybe<Scalars["String"]["input"]>;
};

export type QuerylatestReviewsArgs = {
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
};

export type QueryuserReviewsArgs = {
    username: Scalars["String"]["input"];
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
};

export type Mutation = {
    __typename?: "Mutation";
    addReview: Movie;
    deleteReview: Movie;
};

export type MutationaddReviewArgs = {
    movie_id: Scalars["Int"]["input"];
    username: Scalars["String"]["input"];
    rating: Scalars["Int"]["input"];
    comment: Scalars["String"]["input"];
};

export type MutationdeleteReviewArgs = {
    _id: Scalars["ID"]["input"];
};

export type GetMoviesQueryVariables = Exact<{
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
    filters?: InputMaybe<FiltersInput>;
    sortOption?: InputMaybe<SortingType>;
    search?: InputMaybe<Scalars["String"]["input"]>;
}>;

export type GetMoviesQuery = {
    __typename?: "Query";
    movies: Array<{
        __typename?: "Movie";
        _id: number;
        title: string;
        vote_average: number;
        release_date: Date;
        runtime: number;
        poster_path?: string | null;
    }>;
};

export type GetMovieQueryVariables = Exact<{
    movieId: Scalars["Int"]["input"];
}>;

export type GetMovieQuery = {
    __typename?: "Query";
    movie?: {
        __typename?: "Movie";
        _id: number;
        title: string;
        vote_average: number;
        vote_count: number;
        status: string;
        release_date: Date;
        revenue: number;
        runtime: number;
        backdrop_path?: string | null;
        budget: number;
        homepage?: string | null;
        imdb_id?: string | null;
        original_language: string;
        original_title: string;
        overview: string;
        popularity: number;
        poster_path?: string | null;
        tagline?: string | null;
        genres: Array<string>;
        production_companies: Array<string>;
        production_countries: Array<string>;
        spoken_languages: Array<string>;
        keywords: Array<string>;
        reviews: Array<{
            __typename?: "Review";
            _id: string;
            username: string;
            rating: number;
            comment: string;
            date: Date;
        }>;
    } | null;
};

export type GetFiltersQueryVariables = Exact<{ [key: string]: never }>;

export type GetFiltersQuery = {
    __typename?: "Query";
    filters: {
        __typename?: "Filters";
        Genre: Array<string>;
        Rating: Array<string>;
        Decade: Array<string>;
        Status: Array<string>;
        Runtime: Array<string>;
    };
};

export type AddReviewMutationVariables = Exact<{
    movieId: Scalars["Int"]["input"];
    username: Scalars["String"]["input"];
    rating: Scalars["Int"]["input"];
    comment: Scalars["String"]["input"];
}>;

export type AddReviewMutation = {
    __typename?: "Mutation";
    addReview: {
        __typename?: "Movie";
        reviews: Array<{
            __typename?: "Review";
            _id: string;
            username: string;
            rating: number;
            comment: string;
            date: Date;
        }>;
    };
};
