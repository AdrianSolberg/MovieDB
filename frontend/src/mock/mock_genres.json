[
    {
        "id": 0,
        "name": "Drama"
    },
    {
        "id": 1,
        "name": "Romance"
    },
    {
        "id": 2,
        "name": "Western"
    },
    {
        "id": 3,
        "name": "Animation"
    },
    {
        "id": 4,
        "name": "War"
    },
    {
        "id": 5,
        "name": "Science Fiction"
    },
    {
        "id": 6,
        "name": "Comedy"
    },
    {
        "id": 7,
        "name": "Fantasy"
    },
    {
        "id": 8,
        "name": "Adventure"
    },
    {
        "id": 9,
        "name": "Crime"
    },
    {
        "id": 10,
        "name": "Thriller"
    },
    {
        "id": 11,
        "name": "Family"
    },
    {
        "id": 12,
        "name": "Action"
    },
    {
        "id": 13,
        "name": "Mystery"
    }
]
