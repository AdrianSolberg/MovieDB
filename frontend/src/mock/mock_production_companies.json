[
    "Lynda Obst Productions",
    "Pascal Pictures",
    "Fairview Entertainment",
    "Silver Pictures",
    "Walt Disney Pictures",
    "Visiona Romantica",
    "DC Films",
    "Mandate International",
    "Patalex IV Productions Limited",
    "Amblin Entertainment",
    "Heyday Films",
    "Universal Pictures",
    "Bron Studios",
    "Juno Pix",
    "Color Force",
    "LStar Capital",
    "Miramax",
    "The Saul Zaentz Company",
    "Sikelia Productions",
    "Red Granite Pictures",
    "The Weinstein Company",
    "Knickerbocker Films",
    "Columbia Pictures",
    "Groucho II Film Partnership",
    "Syncopy",
    "P of A Productions Limited",
    "Zehnte Babelsberg Film",
    "The Linson Company",
    "Pixar",
    "Appian Way",
    "The Steve Tisch Company",
    "Genre Films",
    "Paramount",
    "Kennedy Miller Productions",
    "Lucasfilm Ltd.",
    "New Line Cinema",
    "Atman Entertainment",
    "Cecchi Gori Pictures",
    "Castle Rock Entertainment",
    "Isobel Griffiths",
    "Phoenix Pictures",
    "DC Entertainment",
    "Dune Entertainment",
    "20th Century Fox",
    "Marvel Studios",
    "Lightstorm Entertainment",
    "WingNut Films",
    "Fox 2000 Pictures",
    "RatPac Entertainment",
    "Lionsgate",
    "Warner Bros. Pictures",
    "Patalex III Productions Limited",
    "Jersey Films",
    "Joint Effort",
    "Village Roadshow Pictures",
    "EMJAG Productions",
    "The Donners' Company",
    "1492 Pictures",
    "A Band Apart",
    "DC Comics",
    "Atlas Entertainment",
    "Ingenious Media",
    "Legendary Pictures",
    "Taurus Film",
    "Wendy Finerman Productions",
    "Peninsula Films",
    "Lin Pictures",
    "Regency Enterprises",
    "Jerry Bruckheimer Films"
]
