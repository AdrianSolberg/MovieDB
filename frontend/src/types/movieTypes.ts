/* eslint-disable @typescript-eslint/no-empty-object-type */

// We disable the eslint rule above because we want to use empty interfaces to
// differentiate between different types even though they have the same structure

interface IdName {
    id: number;
    name: string;
}

interface Genre extends IdName {}
interface Keyword extends IdName {}
interface Language extends IdName {}
interface ProductionCompany extends IdName {}
interface ProductionCountry extends IdName {}

enum Status {
    Released = "Released",
    InProduction = "In Production",
    PostProduction = "Post Production",
    Planned = "Planned",
    Rumored = "Rumored",
    Canceled = "Canceled",
}

interface Movie {
    id: number; // Unique identifier
    title: string; // Unique in combination with release_date
    vote_average: number;
    vote_count: number;
    status: Status;
    release_date: Date;
    revenue: number;
    runtime: number;
    adult: boolean;
    backdrop_path: string | undefined;
    budget: number;
    homepage: string | undefined;
    imdb_id: string | undefined;
    original_language: string;
    original_title: string;
    overview: string;
    popularity: number;
    poster_path: string | undefined;
    tagline: string | undefined;
    genres: Genre[];
    production_companies: ProductionCompany[];
    production_countries: ProductionCountry[];
    spoken_languages: Language[];
    keywords: Keyword[];
}

export { Status };
export type {
    Genre,
    Keyword,
    Language,
    ProductionCompany,
    ProductionCountry,
    Movie,
};
