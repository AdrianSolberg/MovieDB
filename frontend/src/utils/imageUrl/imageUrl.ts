import imageSizes from "./imageSizes.json";
import default_poster from "@/assets/default_poster.webp";

const baseImageUrl = "https://image.tmdb.org/t/p/";

type SizesDictionary = {
    [key: string]: string[];
};

export enum ImageType {
    BACKDROP = "backdrop",
    LOGO = "logo",
    POSTER = "poster",
    PROFILE = "profiel",
    STILL = "still",
}

export const getImageUrl = (
    type: ImageType,
    imagePath: string | undefined,
    imageSize: string = "original"
): string => {
    if (!imagePath) {
        //we only have a default image for poster, can add more later if needed.
        return type == ImageType.POSTER ? default_poster : "";
    }

    const image_sizes: SizesDictionary = imageSizes;
    const sizes: string[] = image_sizes[type];
    if (!sizes.includes(imageSize)) {
        console.error(
            `Invalid image size: ${imageSize}. The size is on the format "w<width>".`
        );
        return "";
    }
    return `${baseImageUrl}${imageSize}${imagePath}`;
};
