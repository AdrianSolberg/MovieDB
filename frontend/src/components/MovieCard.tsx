import {
    Card,
    CardContent,
    CardDescription,
    CardHeader,
} from "@/shadcn/components/ui/card";
import Ratings from "@/shadcn/components/ui/rating";
import { Movie } from "@/types/movieTypes";
import { Link } from "react-router-dom";

const MovieCard = ({ movie }: { movie: Movie }) => {
    return (
        <Link to={`/movie/${movie._id}`}>
            <Card className="shadow-lg hover:shadow-slate-600 hover:cursor-pointer m-0">
                <CardHeader className="p-2">
                    <img
                        className="w-full"
                        src={
                            "https://image.tmdb.org/t/p/w500" +
                            movie.poster_path
                        }
                        alt={movie.title}
                    />
                </CardHeader>
                <CardContent className="p-2">
                    <CardDescription className="text-center">
                        {movie.release_date.getFullYear()} â€¢{" "}
                        {Math.floor(movie.runtime / 60)}h {movie.runtime % 60}m
                    </CardDescription>
                    <Ratings
                        className="flex justify-around"
                        value={movie.vote_average / 2}
                        variant="yellow"
                        totalstars={5}
                    />
                </CardContent>
            </Card>
        </Link>
    );
};

export default MovieCard;
