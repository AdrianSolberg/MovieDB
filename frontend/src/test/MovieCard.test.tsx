import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import MovieCard from "@/components/MovieCard";
import { MoviePoster } from "@/types/movieTypes";
import { getImageUrl, ImageType } from "@/utils/imageUrl/imageUrl";
import { all_movies } from "./mock/util";
import { vi } from "vitest";
import "@testing-library/jest-dom";

vi.mock("@/utils/imageUrl/imageUrl", () => ({
    getImageUrl: vi.fn(),
    ImageType: {
        POSTER: "poster",
    },
}));

const mockMovie = all_movies[18] as MoviePoster; // Joker

describe("MovieCard", () => {
    const renderMovieCard = () => {
        render(
            <MemoryRouter>
                <MovieCard movie={mockMovie} />
            </MemoryRouter>
        );
    };

    afterEach(() => {
        vi.clearAllMocks();
    });

    it("matches snapshot", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <MovieCard movie={mockMovie} />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders movie card with correct link", () => {
        renderMovieCard();

        expect(screen.getByRole("link")).toHaveAttribute(
            "href",
            "/movie/475557"
        );
    });

    it("renders movie poster with correct attributes", () => {
        vi.mocked(getImageUrl).mockReturnValue("mocked-image-url");

        renderMovieCard();

        const image = screen.getByRole("img");
        expect(image).toHaveAttribute("src", "mocked-image-url");
        expect(image).toHaveAttribute("alt", "Joker");
        expect(image).toHaveAttribute("title", "Joker");

        expect(getImageUrl).toHaveBeenCalledWith(
            ImageType.POSTER,
            "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
            "w342"
        );
    });

    it("displays correct release year and runtime", () => {
        render(
            <MemoryRouter>
                <MovieCard movie={mockMovie} />
            </MemoryRouter>
        );

        expect(screen.getByText("2019")).toBeInTheDocument();
        expect(screen.getByText("2h 2m")).toBeInTheDocument();
    });

    it("renders rating component with correct value", () => {
        renderMovieCard();

        expect(screen.getByTitle("4.08")).toBeInTheDocument();
    });

    it("handles missing poster path", () => {
        const movieWithoutPoster = {
            ...mockMovie,
            poster_path: "",
        };

        render(
            <MemoryRouter>
                <MovieCard movie={movieWithoutPoster} />
            </MemoryRouter>
        );

        expect(getImageUrl).toHaveBeenCalledWith(ImageType.POSTER, "", "w342");
    });
});
