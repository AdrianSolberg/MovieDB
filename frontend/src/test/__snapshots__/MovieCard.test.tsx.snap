// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieCard > matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/movie/475557"
  >
    <div
      class="rounded-lg border bg-card text-card-foreground m-0 shadow-lg hover:cursor-pointer hover:shadow-slate-600"
    >
      <div
        class="flex flex-col space-y-1.5 aspect-[2/3] p-2"
      >
        <img
          alt="Joker"
          class="h-full object-scale-down object-top"
          title="Joker"
        />
      </div>
      <div
        class="p-6 pt-0 px-2 pb-2"
      >
        <p
          class="text-sm text-muted-foreground text-center"
        >
          <time
            datetime="2019"
          >
            2019
          </time>
           â€¢ 
          <time
            datetime="PT2H2M"
          >
            2h 2m
          </time>
        </p>
        <div
          class="flex justify-around"
          title="4.08"
        >
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <svg
            class="lucide lucide-star fill-current text-yellow-500"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
            />
          </svg>
          <div
            class="relative inline-block"
          >
            <svg
              class="lucide lucide-star fill-transparent text-yellow-500"
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <div
              style="position: absolute; top: 0px; overflow: hidden; width: 8.399999999999963%;"
            >
              <svg
                class="lucide lucide-star fill-current text-yellow-500"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
